Adding Docker image build to Travis

giovannicuriel opened the pull request.

giovannicuriel committed:
Adding Docker image build to Travis

giovannicuriel committed:
Update .travis.yml

mmagr commented on .travis.yml:
```diff
@@ -1,8 +1,20 @@
+ - docker build -t dojot/iotagent-json:latest . -f ./docker/Dockerfile
+
+after_success:
+  - if [ "${TRAVIS_BRANCH}" == "cpqd_master" ]; then
```
This will effectively prevent us from using travis to generate artifacts for tags.
On other repos, to keep travis' configurations smaller, we've used a helper [script](https://github.com/dojot/gui/blob/master/travis/publish.sh) to handle the container building part.

giovannicuriel replied:
I'll copy from there.

mmagr commented on .travis.yml:
```diff
@@ -1,8 +1,20 @@
 
 install:
  - npm install
  - npm run-script build
```
shouldn't this be in `script`?

giovannicuriel replied:
Yes.

giovannicuriel committed:
Adjusting Travis configuration

- Moving Dockerfile to root folder (in par with other repos)
- Adding publish to DockerHub script
- Updating .travis.yml to reflect these changes

giovannicuriel committed:
Adding a new line at the end of publish.sh script

ericbaum commented on travis/publish.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -ex
+
+version="latest"
+if [ $TRAVIS_BRANCH != "master" ] ; then
```
In this case, the correct branch would be cpqd_master, correct?

giovannicuriel committed:
Setting publish.sh as executable

mmagr committed:
Changes default source branch
